casper/src/main/scala/coop/rchain/casper/util/comm/CommUtil.scala

```diff
@@ -2,61 +2,62 @@ package coop.rchain.casper.util.comm
 
 import cats.{Applicative, Monad}
 import cats.implicits._
-
 import coop.rchain.casper.MultiParentCasper
 import coop.rchain.casper.protocol.BlockMessage
 import coop.rchain.comm.ProtocolMessage
 import coop.rchain.comm.protocol.rchain.Packet
-import coop.rchain.p2p.effects.{Log, NodeDiscovery, TransportLayer}
+import coop.rchain.p2p.effects._
 import coop.rchain.crypto.codec.Base16
+import coop.rchain.p2p.Network.{frameMessage, ErrorHandler, KeysStore}
+import coop.rchain.p2p.NetworkProtocol
+
 import scala.util.Try
 
 object CommUtil {
-  def sendBlock[F[_]: Monad: NodeDiscovery: TransportLayer: Log](b: BlockMessage): F[Unit] = {
-    val packet = blockMessageToPacket(b)
-    val msg    = framePacket(packet)
-
+  def sendBlock[
+      F[_]: Monad: NodeDiscovery: TransportLayer: Log: Time: Encryption: KeysStore: ErrorHandler](
```

Comment:
Looking at all this type classes required for `F[_]` -- doesn't it break encapsulation principle? Why caller of this should ever by worried by all of it? I should be able to call `sendBlock` with arguments and that's it. I don't care if it's using encryption, whether it logs errors or measure execution time or that it need some `TransportLayer`. I think I prefer pure OO with proper encapsulation than this. Any thoughts? ping @rabbitonweb @henrytill

Reply:
I think the issue is that we have to call frameMessage[F], which ends up pulling all those type classes in. I think TransportLayer[F].commSend should just be able to take a Packet .

Reply:
@goral09 I'm more then happy to have a conversation about it today, after standup?